PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc

```diff
@@ -78,6 +78,35 @@ namespace cms::Ort {
 
   ONNXRuntime::~ONNXRuntime() {}
 
+  SessionOptions ONNXRuntime::defaultSessionOptions(GPUMode gpu_mode) {
+    SessionOptions sess_opts;
+    sess_opts.SetIntraOpNumThreads(1);
+    if (gpu_mode != no_gpu) {
+      // detect if there is GPU
+      int devices = 0;
+      auto status = cudaGetDeviceCount(&devices);
+      bool is_gpu_available = (status == cudaSuccess && devices > 0);
+      if (is_gpu_available) {
+        // https://www.onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html
+        OrtCUDAProviderOptions options;
+        options.device_id = 0;
+        options.arena_extend_strategy = 0;
+        options.cuda_mem_limit = std::numeric_limits<std::size_t>::max();
+        options.cudnn_conv_algo_search = OrtCudnnConvAlgoSearch::EXHAUSTIVE;
+        options.do_copy_in_default_stream = 1;
+        sess_opts.AppendExecutionProvider_CUDA(options);
+      } else {
+        // if GPU is not available
+        if (gpu_mode == force_gpu) {
+          throw cms::Exception("RuntimeError") << "No GPU detected, cannot run ONNXRuntime on GPU.";
```

Comment:
This would be better as
```suggestion
          throw cms::Exception("UnavailableAccelerator") << "No GPU detected, cannot run ONNXRuntime on GPU.";
```
to align with #36699 (the string category should get mapped to `edm::errors::UnavailableAccelerator`) when #36699 gets in).